client/my-sites/plugins/plugins-browser/index.jsx

```diff
@@ -497,27 +477,31 @@ const FullListView = ( {
 } ) => {
 	const translate = useTranslate();
 
-	let plugins = pluginsByCategory;
-	let isFetching = isFetchingPluginsByCategory;
-	if ( category === 'paid' ) {
-		plugins = paidPlugins;
-		isFetching = isFetchingPaidPlugins;
-	}
+	const {
+		data: { plugins = [] } = {},
+		isLoading: isFetching,
+		fetchNextPage,
+	} = useWPORGInfinitePlugins( { category } );
+	const isPaidCategory = category === 'paid';
 
 	if ( plugins.length > 0 || isFetching ) {
 		return (
-			<PluginsBrowserList
-				plugins={ plugins }
-				listName={ category }
-				title={ translateCategoryTitle( { category, translate } ) }
-				site={ siteSlug }
-				showPlaceholders={ isFetching }
-				currentSites={ sites }
-				variant={ PluginsBrowserListVariant.InfiniteScroll }
-				billingPeriod={ billingPeriod }
-				setBillingPeriod={ category === 'paid' && setBillingPeriod }
-				extended
-			/>
+			<>
+				<PluginsBrowserList
+					plugins={ isPaidCategory ? paidPlugins : plugins }
+					listName={ category }
+					title={ translateCategoryTitle( { category, translate } ) }
+					site={ siteSlug }
+					showPlaceholders={ isPaidCategory ? isFetchingPaidPlugins : isFetching }
+					currentSites={ sites }
+					variant={ PluginsBrowserListVariant.InfiniteScroll }
+					billingPeriod={ billingPeriod }
+					setBillingPeriod={ isPaidCategory && setBillingPeriod }
+					extended
+				/>
+
+				<InfiniteScroll nextPageMethod={ fetchNextPage } />
```

Comment:
So at all other instances, if we don't make this callback, the query would just work as one-time fetch. Because of how [the cache key is set up](https://github.com/Automattic/wp-calypso/pull/62519/files#diff-4aa4ee82e625627654fb119928b8c515c079207f6273d88cb27bbf02d1fc3feeR30-R32), these queries would also be cached separately anyway.